### DISCUSSION TITLE
Issue when trying to use the "number of cells connected in series to make a battery" setting in Prada2013

### DISCUSSION THREAD

**Discussion by skywo1f** [pybamm-team/PyBaMM#2561] [General]
I am trying to simulate a system with 16 lfp batteries in series. I modified the Prada2013 file to reflect that: 

 "Number of electrodes connected in parallel to make a cell": 4.0,
        "Number of cells connected in series to make a battery": 16.0,
        "Lower voltage cut-off [V]": 2.5,
        "Upper voltage cut-off [V]": 3.65,
        "Initial concentration in negative electrode [mol.m-3]": 28831.45783,
        "Initial concentration in positive electrode [mol.m-3]": 35.3766672,
        "Initial temperature [K]": 298.15,

Then I set the ranges of my experiment:

exp = pybamm.Experiment(
    [
        (
            "Charge at 1 C until 58.4 V",
            "Hold at 58.4 V until C/10",
            "Rest for 5 minutes",
            "Discharge at 1 C until 40 V",
            "Rest for 5 minutes",
        )
    ]
    )

But it is still complaining:

2022-12-14 15:16:49.165 - [ERROR] callbacks.on_experiment_error(226): Simulation error: Could not find consistent states: Could not find acceptable solution: .../casadi/core/linsol.cpp:173: Assertion "m->is_nfact" failed:

Are there any other parameters in Prada2013 that I need to modify?

**Comment by brosaplanella** [comment]
Is it crashing at the very beginning? It seems that you start with a fully charged battery but the first step is charging. Could be related to that.

**Reply by skywo1f** [reply]
That wasn't an issue when it was only 1 battery

**Reply by skywo1f** [reply]
sorry I also forgot to show the solve line:
sim.solve([0, 3600],initial_soc=0)  # solve for 1 hour

**Comment by skywo1f** [comment]
If I change the experiment and solution to:
```
exp = pybamm.Experiment(
    [
        (
#            "Charge at 1 C until 58.4 V",
#            "Hold at 58.4V until C/10",
            "Rest for 5 minutes",
            "Discharge at 0.01 C until 40 V",
            "Rest for 5 minutes",
        )
    ]
    )
sim.solve([0, 3600])  # solve for 1 hour

```
Then it complains:
2022-12-15 07:22:16.631 - [ERROR] callbacks.on_experiment_error(226): Simulation error: Events ['Maximum voltage'] are non-positive at initial conditions
Traceback (most recent call last):
  File "lfp_sag16.py", line 40, in <module>
    sim.solve([0, 3600])  # solve for 1 hour
  File "/home/iviti/.local/lib/python3.8/site-packages/pybamm/simulation.py", line 746, in solve
    raise error
  File "/home/iviti/.local/lib/python3.8/site-packages/pybamm/simulation.py", line 724, in solve
    step_solution = solver.step(
  File "/home/iviti/.local/lib/python3.8/site-packages/pybamm/solvers/base_solver.py", line 1184, in step
    self._check_events_with_initial_conditions(t_eval, model, ext_and_inputs)
  File "/home/iviti/.local/lib/python3.8/site-packages/pybamm/solvers/base_solver.py", line 1063, in _check_events_with_initial_conditions
    raise pybamm.SolverError(
pybamm.expression_tree.exceptions.SolverError: Events ['Maximum voltage'] are non-positive at initial conditions

**Comment by skywo1f** [comment]
it is odd, but
```
exp = pybamm.Experiment(
    [
        (
#            "Charge at 1 C until 58.4 V",
#            "Hold at 58.4V until C/10",
            "Rest for 5 minutes",
            "Discharge at 0.01 C until 40 V",
            "Rest for 5 minutes",
        )
    ]
    )
sim.solve([0, 3600],initial_soc=1)  # solve for 1 hour 
```
seems to work

**Comment by skywo1f** [comment]
another odd thing is that if I do:
```
solution = sim.solution
A1 = solution["Current [A]"]
t1 = solution["Time [s]"]
V1 = solution["Terminal voltage [V]"]
```

I get ranges in the 2.5-3.65 V, instead of the expected 40-58.4V range (for V1.entries)

I will start a new discussion thread for this line of questions

**Reply by valentinsulzer** [reply]
The variable to use here is "Battery voltage [V]"

**Comment by skywo1f** [comment]
Another odd thing, this doesnt seem to work for Mohtat2020. There I set similar conditions:
```
exp = pybamm.Experiment(
    [
        (
            "Rest for 5 minutes",
            "Discharge at 0.01 C until 43.4 V",
            "Rest for 5 minutes",
        )
    ]
    )
model = pybamm.lithium_ion.SPM()
sim = pybamm.Simulation(model,experiment=exp,parameter_values=param)
sim.solve([0, 3600],initial_soc=1)  # solve for 1 hour
```
and
```
  "Number of cells connected in series to make a battery": 14.0,
        "Lower voltage cut-off [V]": 43.4,
        "Upper voltage cut-off [V]": 58.8,

```
but it tells me:
`ValueError: The upper bound of the voltage, 5.2539V, is less than the target maximum voltage, 58.8000V. Stoichiometry limits are x:[0.0000, 0.8659], y:[0.0000, 0.8925].`

**Reply by valentinsulzer** [reply]
Voltage cut-offs are at the single cell level, not battery
